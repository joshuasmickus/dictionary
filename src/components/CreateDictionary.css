label {
  display: block;
  padding-right: 8px;
}

label span {
  display: inline-block;
  width: 100px;
}
