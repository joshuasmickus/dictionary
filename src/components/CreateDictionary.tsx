import { Form, Formik } from 'formik';
import * as React from 'react';

interface ICreateDictionaryState {
  validated: boolean;
}
const validate = (validater: any) => {
  console.log(validater); // tslint:disable-line
};

export class CreateDictionary extends React.Component<{}, ICreateDictionaryState> {
  public state = {
    validated: false
  };

  public render() {
    return (
      <Formik
        onSubmit={this.handleSubmit}
        initialValues={{
          from: '',
          to: ''
        }}
        validate={validate}
      >
        {({
          // handleSubmit,
          handleChange,
          // handleBlur,
          // values,
          // touched,
          // isValid,
          // errors,
        }) => (
          <Form>
            <h1>Create dictionary item</h1>
            <label>From:</label>
            <input
              name="from"
              required={true}
              type="text"
              onChange={this.handleChange(handleChange)}
            />
            <label>To:</label>
            <input
              name="to"
              required={true}
              type="text"
              onChange={this.handleChange(handleChange)}
            />
            <button type="submit">Create dictionary</button>
        </Form>
        )}
      </Formik>
    )
  }

  private handleChange = (handleChange: Formik['handleChange']) => (event: React.FormEvent) => {
    handleChange(event);
  }
  private handleSubmit = (event: any) => {
     console.log(event); // tslint:disable-line
  }
}
