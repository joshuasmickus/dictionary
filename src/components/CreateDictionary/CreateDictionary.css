label {
  display: block;
  padding-right: 8px;
}

label {
  margin: 12px 12px 12px 0;
  padding: 12px 12px 12px 0;
}

label span {
  display: inline-block;
  width: 100px;
}

span.error {
  border: 0;
  color: red;
  font-size: 10px;
  margin: 0 12px;
  padding: 0 12px;
}
