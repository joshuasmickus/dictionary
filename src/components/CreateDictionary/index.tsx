import { connect } from 'react-redux';

import {
  CreateDictionary,
  mapDispatchToProps,
  mapStateToProps
} from './CreateDictionary';


export default connect(mapStateToProps, mapDispatchToProps)(CreateDictionary);
