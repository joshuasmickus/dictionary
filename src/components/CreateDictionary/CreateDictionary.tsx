import { Form, Formik } from 'formik';
import * as React from 'react';
import { Redirect } from 'react-router-dom';
import { toast } from 'react-toastify';

import { IDictionaryName, IDictionaryState } from 'src/models/common';
import { createDictionary } from 'src/state/dictionary/dictionaryRoutines';

import './CreateDictionary.css';

interface IDictionaryErrors {
  name?: string;
};
interface ICreateDictionaryProps {
  createDictionaryRequest: (values: IDictionaryName) => void;
};
interface ICreateDictionaryState {
  redirectToHome: boolean;
};
const validate = (values: any) => {
  const errors: IDictionaryErrors = {};

  if (values.name === '') {
    errors.name = 'Enter a name';
  }

  return errors;
};

export class CreateDictionary extends React.Component<ICreateDictionaryProps, ICreateDictionaryState> {
  public state = {
    redirectToHome: false
  };

  public render() {
    if (this.state.redirectToHome) {
      return <Redirect to="/" />;
    }
    return (
      <Formik
        onSubmit={this.handleSubmit}
        initialValues={{
          name: ''
        }}
        validate={validate}
      >
        {({
          handleChange,
          handleBlur,
          touched,
          errors,
        }) => (
          <Form>
            <h1>Create dictionary</h1>
            <label>
              <span>Name:</span>
              <input
                name="name"
                required={true}
                type="text"
                onChange={handleChange}
                onBlur={handleBlur}
              />
              {touched.name && errors.name && (<span className="error">{errors.name}</span>)}
            </label>
            <label>
              <span>&nbsp;</span>
              <button type="submit">Create dictionary</button>
            </label>
        </Form>
        )}
      </Formik>
    )
  }

  private handleSubmit = (values: IDictionaryName) => {
     this.props.createDictionaryRequest(values);

     toast(`Dictionary '${values.name}' created!`);

     this.setState({
      redirectToHome: true
     });
  }
}

export const mapStateToProps = ({ dictionary }: IDictionaryState) => ({
  list: dictionary.list
});

export const mapDispatchToProps = (dispatch: (action: any) => void) => ({
  createDictionaryRequest: (values: IDictionaryName) => dispatch(createDictionary.request(values))
});
