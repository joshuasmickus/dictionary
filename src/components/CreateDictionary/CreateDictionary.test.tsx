import { createMemoryHistory } from 'history'
import * as React from 'react';
import { Router } from 'react-router-dom';
import { render } from 'react-testing-library';

import { CreateDictionary } from './CreateDictionary';

const requiredProps = {
  createDictionaryRequest: jest.fn(),
};

describe('Given a CreateDictionary component', () => {
  describe('when it is rendered', () => {
    it('should match the snapshot', () => {
      const { container } = render(<Router history={createMemoryHistory()}><CreateDictionary {...requiredProps} /></Router>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
