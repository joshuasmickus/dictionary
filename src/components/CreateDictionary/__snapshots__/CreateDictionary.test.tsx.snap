// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a CreateDictionary component when it is rendered should match the snapshot 1`] = `
<form>
  <h1>
    Create dictionary
  </h1>
  <label>
    <span>
      Name:
    </span>
    <input
      name="name"
      required=""
      type="text"
    />
  </label>
  <label>
    <span>
      
    </span>
    <button
      type="submit"
    >
      Create dictionary
    </button>
  </label>
</form>
`;
